cmd/lifecycle-show.go

```diff
@@ -0,0 +1,579 @@
+/*
+ * MinIO Client (C) 2020 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"bytes"
+	"errors"
+	"os"
+	"strconv"
+
+	"github.com/fatih/color"
+	"github.com/minio/cli"
+	"github.com/minio/mc/pkg/probe"
+	"github.com/minio/minio/pkg/console"
+)
+
+var ilmShowCmd = cli.Command{
+	Name:   "show",
+	Usage:  "show bucket lifecycle configuration",
+	Action: mainLifecycleShow,
+	Before: setGlobalsFromContext,
+	Flags:  append(ilmShowFlags, globalFlags...),
+	CustomHelpTemplate: `Name:
+	{{.HelpName}} - {{.Usage}}
+
+USAGE:
+ {{.HelpName}} [COMMAND FLAGS] TARGET
+
+FLAGS:
+ {{range .VisibleFlags}}{{.}}
+ {{end}}
+DESCRIPTION:
+  Ilm show command displays current lifecycle configuration.
+
+EXAMPLES:
+1. Show the lifecycle management rules (all fields) for the testbucket on alias s3.
+	{{.Prompt}} {{.HelpName}} s3/testbucket
+
+2. Show the lifecycle management rules (fields related to expration date/days) for the testbucket on alias s3.
+	{{.Prompt}} {{.HelpName}} --expiry s3/testbucket
+
+3. Show the lifecycle management rules (fields related to transition date/days, storage class) for the testbucket on alias s3.
+	{{.Prompt}} {{.HelpName}} --transition s3/testbucket
+
+4. Show the lifecycle management rules (minimum details) for the testbucket on alias s3.
+	{{.Prompt}} {{.HelpName}} --minimum s3/testbucket
+
+5. Show the lifecycle management rules in JSON format for the testbucket on alias s3.
+	{{.Prompt}} {{.HelpName}} --json s3/testbucket
+
+`,
+}
+
+var ilmShowFlags = []cli.Flag{
+	cli.BoolFlag{
+		Name:  "expiry",
+		Usage: "show expiration field(s)",
+	},
+	cli.BoolFlag{
+		Name:  "transition",
+		Usage: "show transition field(s)",
+	},
+	cli.BoolFlag{
+		Name:  "minimum",
+		Usage: "show minimum fields",
+	},
+}
+
+func invalidFlagSet(ctx *cli.Context) bool {
+	var flags = [...]bool{ctx.Bool("expiry"), ctx.Bool("transition"), ctx.Bool("json"), ctx.Bool("transition"), ctx.Bool("minimum")}
+	foundSet := false
+	idx := 0
+	for range flags {
+		if foundSet && flags[idx] {
+			return true
+		} else if flags[idx] {
+			foundSet = true
+		}
+		idx++
+	}
+	return false
+}
+
+// showopts gives an idea about what details user prefers to see.
+func getShowOpts(ctx *cli.Context) showDetails {
+	showOpts := showDetails{
+		expiry:       ctx.Bool("expiry"),
+		transition:   ctx.Bool("transition"),
+		json:         ctx.Bool("json"),
```

Comment:
you need to carry this value `globalJSON` is automatically set when 

> 	Before: setGlobalsFromContext,

is called right before calling ILM command action.